tests/bsim/bluetooth/host/iso/cis/src/cis_central.c

```diff
@@ -0,0 +1,344 @@
+{
+	printk("ISO Channel %p connected\n", chan);
+
+	seq_num = 0U;
+	enqueue_cnt = ENQUEUE_COUNT;
+
+	/* Start send timer */
+	k_work_schedule(&iso_send_work, K_MSEC(0));
+
+	SET_FLAG(flag_iso_connected);
+}
+
+static void iso_disconnected(struct bt_iso_chan *chan, uint8_t reason)
+{
+	printk("ISO Channel %p disconnected (reason 0x%02x)\n", chan, reason);
+
+	k_work_cancel_delayable(&iso_send_work);
+
+	UNSET_FLAG(flag_iso_connected);
+}
+
+static void sdu_sent_cb(struct bt_iso_chan *chan)
+{
+	int err;
+
+	enqueue_cnt++;
+
+	if (!TEST_FLAG(flag_iso_connected)) {
+		/* TX has been aborted */
+		return;
+	}
+
+	err = k_work_schedule(&iso_send_work, K_NO_WAIT);
+	if (err < 0) {
+		FAIL("Failed to schedule TX for chan %p: %d\n", chan, err);
+	}
+}
+
+static void init(void)
+{
+	static struct bt_iso_chan_ops iso_ops = {
+		.connected = iso_connected,
+		.disconnected = iso_disconnected,
+		.sent = sdu_sent_cb,
+	};
+	static struct bt_iso_chan_io_qos iso_tx = {
+		.sdu = CONFIG_BT_ISO_TX_MTU,
+		.phy = BT_GAP_LE_PHY_2M,
+		.rtn = 1,
+		.path = NULL,
+	};
+	static struct bt_iso_chan_qos iso_qos = {
+		.tx = &iso_tx,
+		.rx = NULL,
+	};
+	int err;
+
+	err = bt_enable(NULL);
+	if (err != 0) {
+		FAIL("Bluetooth enable failed (err %d)\n", err);
+
+		return;
+	}
+
+	for (size_t i = 0U; i < ARRAY_SIZE(iso_chans); i++) {
+		iso_chans[i].ops = &iso_ops;
+		iso_chans[i].qos = &iso_qos;
+#if defined(CONFIG_BT_SMP)
+		iso_chans[i].required_sec_level = BT_SECURITY_L2;
+#endif /* CONFIG_BT_SMP */
+	}
+}
+
+static void create_cig(void)
+{
+	struct bt_iso_chan *channels[1];
+	struct bt_iso_cig_param param;
+	int err;
+
+	channels[0] = default_chan;
+	param.cis_channels = channels;
+	param.num_cis = ARRAY_SIZE(channels);
+	param.sca = BT_GAP_SCA_UNKNOWN;
+	param.packing = BT_ISO_PACKING_SEQUENTIAL;
+	param.framing = BT_ISO_FRAMING_UNFRAMED;
+	param.latency = 10U;          /* ms */
+	param.interval = interval_us; /* us */
+
+	err = bt_iso_cig_create(&param, &cig);
+	if (err != 0) {
+		FAIL("Failed to create CIG (%d)\n", err);
+
+		return;
+	}
+}
+
+static void reconfigure_cig(void)
+{
+	struct bt_iso_chan *channels[2];
+	struct bt_iso_cig_param param;
+	int err;
+
+	for (size_t i = 0U; i < ARRAY_SIZE(channels); i++) {
+		channels[i] = &iso_chans[i];
+	}
+
+	/* Test modifying existing CIS */
+	param.cis_channels = channels;
+	param.num_cis = 1U;
+	param.sca = BT_GAP_SCA_UNKNOWN;
+	param.packing = BT_ISO_PACKING_SEQUENTIAL;
+	param.framing = BT_ISO_FRAMING_UNFRAMED;
+	param.latency = 10U;    /* ms */
+	param.interval = 7500U; /* us */
+
+	err = bt_iso_cig_reconfigure(cig, &param);
+	if (err != 0) {
+		FAIL("Failed to reconfigure CIG to new interval (%d)\n", err);
+
+		return;
+	}
+
+	/* Add CIS to the CIG and restore interval to 10ms */
+	param.num_cis = 2U;
+	param.interval = interval_us; /* us */
+
+	err = bt_iso_cig_reconfigure(cig, &param);
+	if (err != 0) {
+		FAIL("Failed to reconfigure CIG with new CIS and original interval (%d)\n", err);
+
+		return;
+	}
```

Comment:
We could also consider testing reconfiguring the cig with num_cis set to 0.

Reply:
That was a good idea - It uncovered another issue :)